packages/core/admin/admin/src/components/NpsSurvey/index.js

```diff
@@ -123,13 +147,88 @@ const NpsSurvey = () => {
     setSurveyIsShown(false);
   };
 
-  // TODO: replace with the proper UI
+  const onSelectRating = (number) => {
+    setShowFeedbackBox(true);
+    setSelectedRating(number);
+  };
+
   return (
-    <Flex gap={2} padding={2}>
-      <Typography>NPS SURVEY</Typography>
-      <Button onClick={handleDismiss}>Dismiss</Button>
-      <Button onClick={handleSubmitResponse}>Submit</Button>
-    </Flex>
+    <Portal>
+      <BannerWrapper hasRadius>
```

Comment:
What do you think about using formik instead and wrapping the whole banner into a form element? I think in the long term this will make validation (e.g. against a yup schema) much better and be more accessible.

I think it would also be nice to render the options as custom radio inputs over buttons.

Wht do you think?

Reply:
@joshuaellis @gu-stav @markkaylor Yes I thought of radio buttons but it involves a lot styling to look like this secondary button, so I thought of keeping it simple. But as you all are saying semantics should be correct, will update it with form.

Reply:
awesome 💪🏼

Reply:
I agree with the form. The radio buttons I can see an argument both ways. Some questions though:
- Are the buttons tabbable?
- Could we use a role attribute on the current buttons to get the semantics https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/radio_role ?

Reply:
@markkaylor With radio-button you would be able to use arrow keys and the active / selected button would derive the active state from the state of the underlying radio box I think. Therefore not tabbable, but still keyboard-selectable. They only look like buttons though - semantically they would be a label I guess.

Using aria could be an option, yeah - but you'd have to replicate the same keyboard behavior as for radio button, so why not use native APIs instead? :)

Reply:
ok yup radio buttons makes sense to me I thought it was just a question on tabbing but it sounds like the native API comes with more interaction than I thought :).